### ISSUE TITLE
SQL in removeDoubleClicks needs improvement

### ISSUE THREAD

**Issue by bozana** [pkp/pkp-lib#9627]
**Describe the bug**
The query from the function PKPTemporaryTotalsDAO::removeDoubleClicks() never finishes for big installation with lots of log file entries. 
S. https://forum.pkp.sfu.ca/t/ojs-3-4-0-4-statistic-problem/85132, but also University Heidelberg reported the same problem.

Thus use an combined index on load_id, context_id and ip column.
Consider also the similar function compileUniqueClicks().

**What application are you using?**
 OJS, OMP, OPS version 3.4.0.x

PRs:
stable-3_4_0:
- https://github.com/pkp/pkp-lib/pull/9629
- https://github.com/pkp/ojs/pull/4147
- https://github.com/pkp/omp/pull/1502
- https://github.com/pkp/ops/pull/617

main:
- https://github.com/pkp/pkp-lib/pull/9640
- https://github.com/pkp/ojs/pull/4152
- https://github.com/pkp/omp/pull/1503
- https://github.com/pkp/ops/pull/618

**Comment by bozana** [comment]
@asmecher and/or @jonasraoni, could (one of) your take a look into the PRs above?
I added the indexes (load_id, context_id, ip(64)) to the usage stats temporary tables. This does have a great impact. (The query now finishes withing 50s for 250000 entries... comparing with not finishing at all because of the timeout i.e. finishing in 24h as the user from the forum said).
Once these PRs are merged, I would ask affected users to apply the patch, so that their jobs do not fail any more. Because of that -- because some users will have the indexes already before the next release -- I am checking if the indexes exist in the upgrade script (which we else do not do). OK?
Unfortunately, when testing I found another bug :see_no_evil: in OMP -- I have not considered the inheritance for the insert function in two classes correctly back then. Now I corrected it too.
Thanks a lot!
